core/crypto.mk

```diff
@@ -228,3 +228,46 @@ _CFG_CORE_LTC_HASH := $(call ltc-one-enabled, MD5 SHA1 SHA224 SHA256 SHA384 \
 _CFG_CORE_LTC_MAC := $(call ltc-one-enabled, HMAC CMAC CBC_MAC)
 _CFG_CORE_LTC_CBC := $(call ltc-one-enabled, CBC CBC_MAC)
 _CFG_CORE_LTC_ASN1 := $(call ltc-one-enabled, RSA DSA ECC)
+
+###############################################################
+# platform independent SE050 specifics
+###############################################################
+ifeq ($(CFG_NXP_SE05X), y)
+CFG_CORE_SE05X_SCP03_PROVISION ?= n
+CFG_CORE_SE05X_DISPLAY_INFO ?= y
+CFG_CORE_SE05X_SCP03_EARLY ?= y
+CFG_CORE_SE05X_INIT_NVM ?= n
+CFG_CORE_SE05X_OEFID ?= n
+
+# I2C bus configuration
+CFG_CORE_SE05X_BAUDRATE ?= 3400000
+CFG_CORE_SE05X_I2C_BUS ?= 2
+
+# Extra stacks required to support the Plug and Trust external library
+CFG_STACK_THREAD_EXTRA ?= 8192
+CFG_STACK_TMP_EXTRA ?= 8192
+
+CFG_NXP_SE05X_HUK_DRV ?= y
+
+CFG_NXP_SE05X_RNG_DRV ?= y
+ifeq ($(CFG_NXP_SE05X_RNG_DRV), y)
+CFG_WITH_SOFTWARE_PRNG = n
+endif
+
+cryphw-one-enabled = $(call cfg-one-enabled, \
+                        $(foreach v,$(1), CFG_NXP_SE05X_$(v)_DRV))
+
+# Asymmetric ciphers
+CFG_NXP_SE05X_RSA_DRV ?= y
+ifeq ($(CFG_NXP_SE05X_RSA_DRV), y)
+$(call force, CFG_CRYPTO_DRV_RSA, y)
+CFG_CRYPTO_RSASSA_NA1 = n
```

Comment:
sorry I dont understand....why?

Reply:
value `n` is mandated here? if so, use `call force`.
If it is optional, use `?= n`.
Unless I missed something...

Reply:
ah, you probably mean.....(call force,CFG_CRYPTO_RSASSA_NA1,n) not SE05X_RSA_DRV....
ok

Reply:
Yeah! sorry, I meant `CFG_CRYPTO_RSASSA_NA1`

From the issue you report, that means somewhere `CFG_CRYPTO_RSASSA_NA1` is set to an empty value? I think this points a issue in the default config. From the current source tree, I don't see where such empty value is assigned to `CFG_CRYPTO_RSASSA_NA1`. Maybe in your build env?

Reply:
> CFG_CRYPTO_RSASSA_NA1 is set to '' (from undefined) but its value must be ' n'. Stop

That is weird indeed. `_force` in `mk/checkconf.mk` has `ifdef $(1)` so I don't see how `$(origin $(1))` can return "undefined". With the suggested modification, I get this:
```
$ make out/arm-plat-vexpress/conf.mk CFG_NXP_SE05X=y
core/crypto.mk:264: *** CFG_CRYPTO_RSASSA_NA1 is set to 'y' (from file) but its value must be 'n'.  Stop.
```
...which makes sense due to `CFG_CRYPTO_RSASSA_NA1 ?= y` in `mk/config.mk:556`.

Reply:
yes I can see this now. So how do we address this?
```
ifeq ($(CFG_CRYPTO_RSASSA_NA1), y)                                                                                                                                                           
#Tomcrypt enables NA1 it by defaut                                                                                                                                                           
CFG_CRYPTO_RSASSA_NA1=n                                                                                                                                                                      
else                                                                                                                                                                                         
$(call force, CFG_CRYPTO_RSASSA_NA1,n)                                                                                                                                                       
endif 
``` 

or just as I had before:
`CFG_CRYPTO_RSASSA_NA1=n
`

Reply:
`CFG_CRYPTO_RSASSA_NA1 ? = y` should be moved from `mk/config.mk` to `core/crypto.mk` after your code block, because it just sets a default value. The 'force' macro is used when you want to make sure a certain value is used. It does NOT override the value, instead it checks for conflicts: it reports if a different value was requested already (for example, from a precedent Makefile, from the command line or from an environment variable). `?=` sets the value if the variable is unset. Plain `=` is normally not used for `CFG_*`.
https://optee.readthedocs.io/en/latest/building/gits/optee_os.html#configuration-variables

So the correct sequence would be:

```make
ifeq ($(CFG_NXP_SE05X_RSA_DRV),y)
$(call force,CFG_CRYPTO_RSASSA_NA1,n)
endif
CFG_CRYPTO_RSASSA_NA1 ?= y
```

Reply:
of course. ok.